ui/app/common/registrationNaturalFilter/registrationNaturalFilter.js

```diff
@@ -0,0 +1,89 @@
+angular.module("naturalSorting", [])
+
+// The core natural service
+.factory("naturalService", ["$locale", function ($locale) {
```

Comment:
@krishnanspace shouldn't we be adding unit tests for this service?

I release this was implemented as a high priority, but is there a plan or a note to get back to this at some point?

Reply:
@nthfloor The above service is actually an external library I had used for Natural Sort.

Reply:
What do you mean by an external library? From what I can see this is a custom service you are adding to be used by the Bahmni app. So what is making this external and why does that mean we should not unit test it here?

Reply:
We should be unit testing it. Thats for sure.What I meant was that I had not written the service

Reply:
ok, and is this something is being contributed back to the community?

Reply:
Yes, I can see this as a contribution.

Reply:
Ok then I will create a new ticket to add some unit tests and work on contributing this back to the community. Sound good?

Reply:
Yup!